sdks/java/core/src/test/java/org/apache/beam/sdk/testing/PAssertTest.java

```diff
@@ -386,7 +386,7 @@ private static Throwable runExpectingAssertionFailure(Pipeline pipeline) {
     // is first caught by JUnit and causes a test failure.
     try {
       pipeline.run();
-    } catch (AssertionError exc) {
+    } catch (Exception exc) {
```

Comment:
I also don't think it'll break, if it catches an exception/throwable instead since it still asserts the expected message and it'll have to be a very weird coincidence to match a completely different error.